design/ceph-config-updates.md

```diff
@@ -0,0 +1,127 @@
+convenience if they would otherwise be set via command line for every single daemons.
+
+For example:
+- `log file = /dev/stderr` should be set for all daemons to enforce the container standard of
+  sending logging to stdout/stderr.
+- `mon cluster log file = /dev/stderr` also for reasons above
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+  - TODO: should this be set via Ceph centralized config if someone wants to change this? What about Luminous?
+- TODO: did I miss any?
+- `fatal signal handlers = false`
+  **TODO MORE INFORMATION NEEDED** why does Rook set this? It's purpose will determine its config location
+
+#### Additional configs for test environments
+
+Test (especially integration tests) may need to specify `osd pool default size = 1` and
+`osd pool default min size = 1` to support running clusters with only one osd. Test environments
+would have a means of doing this fairly easily using the still-existent config override capability.
+
+### Minimal command line options
+
+The minimal set of command line options is pared down to the settings which inform the daemon where
+to find and/or store data.
+
+Required flags:
+- `--keyring` may necessary to inform daemons where to find their keyrings
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, and these will continue to need
+  set since the locations Rook uses are non-default
+- `--run-dir` will need set on the command line since it is non-default and no longer set in the
+  config file
+
+Notable non-required flags:
+- `--fsid` does not need to be set, as daemons can learn this from their mons
+- `--{public,cluster}-addr` can also be learned from the mons
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config.
+
+Existing global configs:
+- `mon allow pool delete = true`
+- `fatal signal handlers = false` here
+  **TODO** unless it is required by virtue of the container context as with `log file`
+
+### Changes to Ceph mons
+
+The current method of starting mons where `mon-a` has an initial member of `a`, `mon-b` initial
+members `a b`, `mon-c` initial members `a b c`, etc. Has worked so far but could result in a race
+condition. Mon cluster stability is important to Ceph, and it is critical for this PR that the mons'
+centrally-stored config is stable, so we here note that this behavior should be fixed such that the
+mon initial members are known before the first mon is bootstrapped to consider this proposal's work
+completed. Practically speaking, this will merely require the mon services to be started and have
+IP addresses before the mons are bootstrapped.
+
+Additionally, generating the monmap during mon daemon initialization is unnecessary if `mon host` is
+configured and known by `ceph-mon --mkfs` for the entire list of mons.
```

Comment:
@travisn I am inclined to keep a map of all active mons as we currently do in a ConfigMap since I think this will be easier to handle than handling the single seed mon in the case where it has failed. Thoughts?

Reply:
Yes, keeping the list of the expected active mons in the ConfigMap works well. Otherwise we would have to rely on the list of mon services or deployments, which aren't accurate since they may be in some transition state if the operator decided to start a new mon, etc.

Separately, are you saying that if the list of mons changes, `ceph-mon --mkfs` with the new list of mons will reconcile the list of mons?

Reply:
I think what I wrote wasn't very clear. If the `mon host` value is known to the `ceph-mon --mkfs` command, the previous step of generating the `monmap` is unnecessary.